core/user_settings.py

```diff
@@ -75,3 +76,138 @@ def append_to_csv(filename: str, rows: dict[str, str]):
             writer.writerow([])
         for key, value in rows.items():
             writer.writerow([key] if key == value else [value, key])
+
+
+def get_key_value_pairs_and_spoken_forms_from_three_column_csv(
+    filename: str, headers: tuple[str, str, str]
+):
+    """Retrieves a list from a CSV of the form name,values,spoken_forms"""
+    path = compute_csv_path(filename)
+
+    rows = _obtain_rows_from_csv(path)
+
+    result = _convert_rows_from_file_with_headers_to_key_value_pairs_and_spoken_forms(
+        rows, filename, headers
+    )
+    return result
+
+
+def create_three_columns_csv_from_default_if_nonexistent(
+    filename: str,
+    headers: tuple[str, str, str],
+    default: list[list[str, tuple[str], tuple[str]]],
+):
+    path = compute_csv_path(filename)
+    if not path.is_file():
+        _create_three_columns_csv_from_default(path, headers, default)
+
+
+def _create_three_columns_csv_from_default(path, headers, default):
+    with open(path, "w", encoding="utf-8", newline="") as file:
+        writer = csv.writer(file)
+        writer.writerow(headers)
+        for row_tuple in default:
+            row = _compute_row_for_three_column_csv(row_tuple)
+            writer.writerow(row)
+
+
+def _compute_row_for_three_column_csv(input_tuple):
+    if len(input_tuple) == 3:
+        name, values, spoken_forms = input_tuple
+    else:
+        name, values = input_tuple
+        spoken_forms = None
+    values_text = _compute_values_packed_into_column(values)
+    row = [name, values_text]
+    if spoken_forms:
+        spoken_forms_text = _compute_values_packed_into_column(spoken_forms)
+        row.append(spoken_forms_text)
+    return row
+
+
+def _compute_values_packed_into_column(values):
+    output = io.StringIO()
+    writer = csv.writer(output, delimiter=";")
+    writer.writerow(values)
+    result = output.getvalue().strip()
+    return result
+
+
+def _obtain_rows_from_csv(path):
+    with open(str(path), "r") as f:
```

Comment:
Please use `open(str(path), newline='')` to benefit from the standard newline handling.

Reply:
I made the requested change.